core/chains/evm/config/toml/config.go

```diff
@@ -294,14 +294,17 @@ func (c *EVMConfig) ValidateConfig() (err error) {
 	} else if c.ChainID.String() == "" {
 		err = multierr.Append(err, commonconfig.ErrEmpty{Name: "ChainID", Msg: "required for all chains"})
 	} else if must, ok := ChainTypeForID(c.ChainID); ok { // known chain id
-		if c.ChainType == nil && must != "" {
-			err = multierr.Append(err, commonconfig.ErrMissing{Name: "ChainType",
-				Msg: fmt.Sprintf("only %q can be used with this chain id", must)})
-		} else if c.ChainType != nil && *c.ChainType != string(must) {
-			if *c.ChainType == "" {
-				err = multierr.Append(err, commonconfig.ErrEmpty{Name: "ChainType",
-					Msg: fmt.Sprintf("only %q can be used with this chain id", must)})
-			} else if must == "" {
+		is := config.ChainTypeNone
+		var parseErr error
+		if c.ChainType != nil {
+			is, parseErr = config.ChainTypeFromSlug(*c.ChainType)
```

Comment:
Ignored error

Reply:
The error is checked in ln306

Reply:
But it is not referenced after the `if` statement. It seems like it should be handled separately in order to wrap this original error.